### PR TITLE
Refactored card separator

### PR DESCRIPTION
**Pull request by T-Damer**
- Reduced number of available colours
- Shortened props

### CODE REVIEW

File: src/components/CardSeparator.tsx [hunk 1]
```diff
@@ -10,7 +11,27 @@ import classnames, {
...
+  number: number
+  gradient: Gradient
+  vertical?: boolean
+  customHeight?: THeight
+}
+
+function getTailwindClassnames(gradient: Gradient) {
+  switch (gradient) {
+    case 'secondary-to-transparent':
+      return 'from-secondary'
```
Comment (T-Damer) [inline src/components/CardSeparator.tsx:28]: I found out, that tailwind applies transparent if we don't specify `to` or `from`, this is why I return only one string here

File: src/components/CardSeparator.tsx [hunk 2]
```diff
@@ -10,7 +11,27 @@ import classnames, {
...
+    case 'accent-to-secondary':
+      return { 'from-accent': true, 'to-secondary': true }
```
Comment (T-Damer) [inline src/components/CardSeparator.tsx:30]: And this is a way to use two tailwind properties, we can't use an array unfortunately, only `{ prop: true, prop2: true}`. We did it previously in some places, it lookes like:
```TS
gradient(color === 'green' ? { 'from-blue': true, 'to-green': true } : undefined)
```

File: src/components/CardSeparator.tsx [hunk 3]
```diff
@@ -10,7 +11,27 @@ import classnames, {
   width,
 } from 'classnames/tailwind'
 
-type Color = 'accent' | 'primary' | 'secondary' | 'tertiary' | 'transparent'
+type Gradient =
+  | 'secondary-to-transparent'
+  | 'accent-to-secondary'
+  | 'accent-to-transparent'
+interface CardSeparatorProps {
```
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:18]: Don't put interfaces at the top of the file, if you only use them once — put them where you use them

File: src/components/CardSeparator.tsx [hunk 4]
```diff
@@ -10,7 +11,27 @@ import classnames, {
...
+  number: number
```
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:19]: What's `number`? Make the name of the parameter less ambiguous

File: src/components/CardSeparator.tsx (hunk 3 above)
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:18]: Don't forget about the empty lines between declarations

File: src/components/CardSeparator.tsx [hunk 5]
```diff
@@ -10,7 +11,27 @@ import classnames, {
...
+  gradient: Gradient
+  vertical?: boolean
+  customHeight?: THeight
+}
+
+function getTailwindClassnames(gradient: Gradient) {
```
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:25]: Use `classnames` here or whatever function needed from `tailwind-classnames`

File: src/components/CardSeparator.tsx [hunk 6]
```diff
@@ -21,60 +42,19 @@ const connectiveBlock = (vertical = true) =>
...
+export default function (props: CardSeparatorProps) {
   return (
-    <div className={connectiveBlock(vertical)}>
-      {[...Array(number).keys()].map((_, index) => (
-        <div
-          key={index}
-          className={connectiveLine(from, to, vertical, customHeight)}
-        ></div>
+    <div className={connectiveBlock(props.vertical)}>
+      {[...Array(props.number).keys()].map((_, index) => (
```
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:56]: `Array(props.number).fill(null).map(...)`

File: src/components/CardSeparator.tsx (hunk 5 above)
Comment (backmeupplz) [reply src/components/CardSeparator.tsx:25]: Also, name of this function implies that you're getting all the tailwind classnames, you should specify which classnames you're getting here

**Review by backmeupplz — CHANGES_REQUESTED** [review]

File: src/components/CardSeparator.tsx [hunk 7]
```diff
@@ -10,7 +11,23 @@ import classnames, {
...
 } from 'classnames/tailwind'
 
-type Color = 'accent' | 'primary' | 'secondary' | 'tertiary' | 'transparent'
+type Gradient =
+  | 'secondary-to-transparent'
+  | 'accent-to-secondary'
+  | 'accent-to-transparent'
+
+const getTailwindGradient = (gradient: Gradient) =>
+  classnames(
```
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:20]: If it's just one type of classname, you can drop the `classnames`

File: src/components/CardSeparator.tsx [hunk 8]
```diff
@@ -21,61 +38,29 @@ const connectiveBlock = (vertical = true) =>
...
-  from?: Color,
-  to?: Color,
-  vertical = true,
-  customHeight?: number
-) =>
+interface CardSeparatorProps {
+  numberOfLines: number
+  gradient: Gradient
+  vertical?: boolean
+  customHeight?: THeight
```
Comment (backmeupplz) [inline src/components/CardSeparator.tsx:45]: Are you still using this prop?

**Review by backmeupplz — CHANGES_REQUESTED** [review]

File: src/helpers/bigintConvert.ts [hunk 9]
```diff
@@ -98,7 +98,7 @@ export function bnToHex(stringNum: string) {
   return '0x' + hex
 }
 
-function bitnot(bigIntNum: BigInt) {
+function bitnot(bigIntNum: bigint) {
```
[the remark this answers is not in the document]
Comment (T-Damer) [reply src/helpers/bigintConvert.ts:101]: Yes, GH actions can't pass the lint phase

**Review by backmeupplz — APPROVED** [review]

**Pull request merged**